tests/test_initialization.py

```diff
@@ -230,3 +230,27 @@ def test_lora_conv2d_false(self):
         # with init_lora_weights=False, weight B should *not* be zero. We don't care so much about the actual values
         # as long as they are not zero, in order to avoid identity transformation.
         self.assertFalse(torch.allclose(weight_B, torch.zeros_like(weight_B)))
+
+    def test_lora_use_rslora(self):
+        # default is True
+        torch.manual_seed(0)
+
+        _model = self.get_model()
+
+        # check scaling factor use_rslora=False
+        config = LoraConfig(target_modules=["linear", "embed", "conv2d"], lora_alpha=3, r=16, use_rslora=False)
```

Comment:
Thanks for testing the base setting with `use_rslora=False` too. Let's split this into two separate tests, say `test_lora_scaling_default` and `test_rslora_scaling` for better readability. Also, since the `_model` is mutated by `get_peft_model`, it's also the cleaner solution.

On top, could you please also add a test using `rank_pattern` and `alpha_pattern`, just to be sure that this works with rslora as well? So e.g. using `LoraConfig(..., rank_pattern={"embed": 8, "conv2d": 32}, alpha_pattern={"linear": 4, "conv2d": 20})`.